README.md

```diff
@@ -195,52 +195,78 @@ Install custom reporters as a development dependency and you can use the `--repo
+in the coverage report and contribute to coverage statistics.  
+
+Nyc will cover all files if the `--all` flag is set.  In this case all files will
+appear in the coverage report and contribute to coverage statistics.
+
+Nyc will only cover files that are located under `cwd`, and then only `*.js` files
+or files with extensions listed in in the `nyc.extension` array.
+
+You can reduce the set of covered files by adding `include` and `exclude` 
+filter arrays to your config.  These allow you to shape the set of covered files 
+by specifying glob patterns that can filter files from  the covered set.  The `exclude` 
+array may also use exclude negated glob patterns, these are specified with 
+a `!` prefix, and can restore sub-paths of excluded paths.
 
 Globs are matched using [minimatch](https://www.npmjs.com/package/minimatch).
 
-For example, the following config will exclude any files with the extension `.spec.js`,
-and anything in the `build` directory:
+We use the following process to remove files from consideration,  
+ * First, limit the set of covered files to those files in paths listed in the 
+   `include` array.  
+ * Then, remove any files that are found in the `exclude` array.
+ * Finally, restore any exclude negated files that have been excluded in the 
+   second step
 
-```json
-{
-  "nyc": {
-    "exclude": [
-      "**/*.spec.js",
-      "build"
-    ]
-  }
-}
-```
-> Note: Since version 9.0 files under `node_modules/` are excluded by default.
-  add the exclude rule `!**/node_modules/` to stop this.
 
-> Note: exclude defaults to `['coverage/**', 'test/**', 'test{,-*}.js', '**/*.test.js', '**/__tests__/**', '**/node_modules/**']`,
+##### Using include and exclude arrays
+
+If there are paths specified in the `include` array, then the set of 
+covered files will be limited to eligible files found in those paths.
+If the `include` array is left empty then all eligible files will be
+included, equivalent to setting `include` to `['**']`.  Include options
+can be specified on the command line with the `-n` switch.
+
+If there are paths specified in the `exclude` array, then the set of 
+covered files will not feature eligible files found in those paths.
+You can also specify negated paths in the `exclude` array, by prefixing
+them with a `!`.  Negated paths can restore paths that have been 
+already been excluded in the `exclude` array.  Exclude options can be
+specified on the command line with the `-x` switch.
+
+> Note: Exclude defaults to `['coverage/**', 'test/**', 'test{,-*}.js', '**/*.test.js', '**/__tests__/**', '**/node_modules/**']`,
 which would exclude `test`/`__tests__` directories as well as `test.js`, `*.test.js`,
 and `test-*.js` files. Specifying your own exclude property overrides these defaults.
 
-## Including files
+> Note: Since version 9.0, files under `node_modules/` are always excluded by nyc.
+  To reverse this you must add the negated exclude rule `!**/node_modules/`.
```

Comment:
you also have to add back all the other defaults if you want them

Reply:
The note above mentions that exclude will override the defaults, is that enough?

Reply:
@AndrewFinlay See istanbuljs/istanbuljs#213 with regard to this.  Once this is merged we'll want to make NYC support the new option and document that as the preferred way since correctly processing negated excludes can be costly for the `--all` option.

Reply:
@JaKXz I've modified the preceding section to 'hopefully' handle this issue.
@coreyfarrell Sounds like this will have to be expanded on when that drops.  I figure this is a little broken at the moment anyway so it could be shelved until it's fixed.